share/picongpu/tests/Pusher/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,325 @@
+magnetic field. If the relative change between two periods is greater than 
+epsilon = 1e-5, the test fails. Also the absolute phasechange during one turn
+is regarded. It should be smaller than 0.2 rad.
+"""
+
+import openpmd_api as io
+import numpy as np
+import sys
+
+
+def get_params(path):
+    """creates a series of the data found at path.
+    path should have the form data_%T to provide access to all iterations.
+
+    returns:
+        series
+        parameter list... list of parameters of the simulation (units etc.)
+        """
+
+    # load series
+    series = io.Series(path, io.Access.read_only)
+
+    if len(series.iterations) == 1:
+        print("There is just 1 iteration in the series")
+        print("make sure, there are at least two")
+        longer_series = False
+
+    else:
+        longer_series = True
+
+    # read order assignment function
+    i = series.iterations[0]
+
+    # read parameters of the simulation
+    cell_depth = i.get_attribute("cell_depth")
+    cell_height = i.get_attribute("cell_height")
+    cell_width = i.get_attribute("cell_width")
+    dt = i.get_attribute("dt")
+    unit_time = i.get_attribute("unit_time")
+    unit_length = i.get_attribute("unit_length")
+    unit_charge = i.get_attribute("unit_charge")
+    unit_bfield = i.get_attribute("unit_bfield")
+    unit_speed =  i.get_attribute("unit_speed")
+    unit_mass = i.get_attribute("unit_mass")
+
+    return longer_series, series, [
+        cell_depth, cell_height, cell_width, dt, unit_time, unit_charge,
+        unit_length, unit_mass, unit_speed, unit_bfield
+        ]
+
+def read_series(series):
+    """reads the position_offset (grid_knot) and position
+    (float in the intervall [0,1), relativ position in the cell) and the
+    momentum of the particle in the x-y-plane.
+
+    returns:
+        x/y_poss... array of all relative in-Cell positions
+        x/y_offSet... array of all knot-positions
+        x/y_momentum ... array of all momenta
+        charge... charge of particle
+        mass... mass of particle
+        num_iter... number of iterations in the provided series
+        """
+
+    num_iter = len(series.iterations)
+
+    # create arrays for saving the data
+
+    # in-cell position
+    x_poss = np.zeros(num_iter)
+    y_poss = np.zeros(num_iter)
+
+    # cells
+    x_offSet = np.zeros(num_iter)
+    y_offSet = np.zeros(num_iter)
+
+    # momentum
+    x_momentum = np.zeros(num_iter)
+    y_momentum = np.zeros(num_iter)
+
+    i = series.iterations[0]
+    # charge
+    charge = i.particles["e"]["charge"][io.Mesh_Record_Component.SCALAR]
+    # mass
+    mass = i.particles["e"]["mass"][io.Mesh_Record_Component.SCALAR]
+    
+    # for calculation of the openPMD-periode for later use
+    indices = []
+    for iteration_index in series.iterations:
+        indices.append(iteration_index)
+
+    periode = indices[1] -indices[0]
+
+    for iteration in range(0,num_iter,1):
```

Comment:
since there might be no fixed period, could you use `enumare` instead? Or would this break the logic of your code?

Reply:
I now used the indices array for that.